src/contact.rs

```diff
@@ -860,12 +860,14 @@ impl Contact {
         );
         ensure!(origin != Origin::Unknown, "Missing valid origin");
 
-        if context.is_self_addr(addr).await? {
+        if context.is_configured().await? && context.is_self_addr(addr).await? {
             return Ok((ContactId::SELF, sth_modified));
         }
 
-        if !fingerprint.is_empty() {
-            let fingerprint_self = self_fingerprint(context).await?;
+        if !fingerprint.is_empty() && context.is_configured().await? {
+            let fingerprint_self = self_fingerprint(context)
+                .await
+                .context("Failed to get self fingerprint")?;
```

Comment:
I think, to make logs more concise, it's sufficient to just add a function name, e.g. `.context("self_fingerprint")`. At the low level an error usually contains "Failed to..." ot smth like this anyway.